### PR TITLE
#68 add pollIntervalSeconds isNaN validation

### LINKED ISSUE #68: Passing NaN as pollIntervalSeconds interval causes setTimeout infinity call
### Describe the bug
Passing NaN as pollIntervalSeconds interval causes setTimeout infinity call, that in turn millions requests to config-cat CDN

### To reproduce
Pass undefined casted to Number as parameter
```javascript
const myConfig = new Map();
myConfig.set('pollIntervalSeconds', undefined);
new AutoPollOptions("APIKEY", "common", "1.0.0", { pollIntervalSeconds: +(myConfig.get('pollIntervalSeconds')) }, null);

```
### Expected behavior
Configcat client should validate this wrong parameter and throw an exception

Here is recursive with 0 seconds duration
https://github.com/configcat/common-js/blob/master/src/AutoPollConfigService.ts#L91

### PR DESCRIPTION
**Pull request by ddemydenko**
add maxInitWaitTimeSeconds isNaN validation
FIx issue 68
### Describe the purpose of your pull request

Passing NaN as pollIntervalSeconds interval causes setTimeout infinity call, that in turn million request to config-cat CDN

### Related issues (only if applicable)

Provide links to issues relating to this pull request
https://github.com/configcat/common-js/issues/68
### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [x] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

**Comment by endret** [comment]
Hi @ddemydenko,


If you try to pass `null` or `undefinied` to the `pollIntervalSeconds` variables the following [lines](https://github.com/configcat/common-js/blob/master/src/ConfigCatClientOptions.ts#L147) should handle this and the default value will use (60 seconds).

If you agree I close this PR.

Regards,
Endre

**Comment by ddemydenko** [comment]
> Hi @ddemydenko,
> 
> If you try to pass `null` or `undefinied` to the `pollIntervalSeconds` variables the following [lines](https://github.com/configcat/common-js/blob/master/src/ConfigCatClientOptions.ts#L147) should handle this and the default value will use (60 seconds).
> 
> If you agree I close this PR.
> 
> Regards, Endre

Hi, @endret the NaN is not undefined neither null. I faced with this on production and got an unexpected billing.
When you write code with typescript it is ok. But then it compiles to javascript and uses in docker container. So when for some reason(typo mistaken) the env variable was not assigned it caused. to poling with 0 timeout duration.
Here is an example:

```typescript
const configCatConfig: ConfigCatProviderConfig = {
      sdkKey: config.get('CONFIG_CAT_SDK_KEY'),
      pollIntervalSeconds: +config.get('POLL_INTERVAL_SECONDS'),
};
const ConfigCatClient = createClient(configCatConfig);
```

During deployment POLL_INTERVAL_SECONDS was not assigned. This is env vars so it is not under control of typescript code.

`+undefined` will converted to NaN as result poling without duration.

**Comment by endret** [comment]
Thank you for the clarification. Now I see your issue.
I have some concerns with your solution. It is not handled when you pass a `boolean` or `string` with a whitespace value.

I guess the `typeof` or the `.isFinite`could be good for it.

Consider implementing the mentioned solution above, and please create a proper test for the possible cases.

**Comment by ddemydenko** [comment]
> Thank you for the clarification. Now I see your issue. I have some concerns with your solution. It is not handled when you pass a `boolean` or `string` with a whitespace value.
> 
> I guess the `typeof` or the `.isFinite`could be good for it.
> 
> Consider implementing the mentioned solution above, and please create a proper test for the possible cases.

@endret done

File: test/ConfigCatClientOptionsTests.ts [hunk 1]
```diff
@@ -106,6 +106,38 @@ describe("Options", () => {
     }).to.throw("Invalid 'pollIntervalSeconds' value");
   });
 
+  it("AutoPollOptions initialization With NaN 'pollIntervalSeconds' ShouldThrowError", () => {
+    const myConfig = new Map();
+    myConfig.set('pollIntervalSeconds', undefined);
```
Comment (endret) [inline test/ConfigCatClientOptionsTests.ts]:
```suggestion
    myConfig.set('pollIntervalSeconds', NaN);
```

File: test/ConfigCatClientOptionsTests.ts [hunk 2]
```diff
@@ -106,6 +106,38 @@ describe("Options", () => {
...
+    expect(() => {
+      new AutoPollOptions("APIKEY", "common", "1.0.0", { pollIntervalSeconds: +(myConfig.get('pollIntervalSeconds')) }, null);
```
Comment (endret) [inline test/ConfigCatClientOptionsTests.ts]:
```suggestion
      new AutoPollOptions("APIKEY", "common", "1.0.0", { pollIntervalSeconds: myConfig.get('pollIntervalSeconds') }, null);
```

**Review by endret — CHANGES_REQUESTED** [review]

**Review by endret — APPROVED** [review]

**Pull request merged**